src/index.ts

```diff
@@ -454,19 +458,29 @@ export default class AvalancheApp {
     }, processErrorResponse)
   }
 
-  private async _pubkey(path: string, show: boolean, hrp?: string, chainid?: string): Promise<ResponseAddress> {
+  private async _pubkey(path: string, show: boolean, curve_type: number, hrp?: string, chainid?: string): Promise<ResponseAddress> {
```

Comment:
This will not be backward compatible for JS clients that were using hrp and or chainid already. We should move the curve_type to the end of the fn signature.

Reply:
This one is private, but I changed it where this is called because I had it wrong there too.